src/ServiceManagement/RemoteApp/Commands.RemoteApp/Collection/GetAzureRemoteAppCollectionUsageDetails.cs

```diff
@@ -0,0 +1,125 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Management.RemoteApp;
+using Microsoft.Azure.Management.RemoteApp.Models;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Management.Automation;
+using System.Net;
+
+namespace Microsoft.Azure.Management.RemoteApp.Cmdlets
+{
+
+    [Cmdlet(VerbsCommon.Get, "AzureRemoteAppCollectionUsageDetails"), OutputType(typeof(string))]
+    public class GetAzureRemoteAppCollectionUsageDetails : RdsCmdlet
+    {
+        [Parameter(
+	               Position = 0,
+				   Mandatory = true,
+                   ValueFromPipelineByPropertyName = true,
+                   HelpMessage = "RemoteApp collection name")]
+        public string CollectionName { get; set; }
+
+        [Parameter(Mandatory = false,
+            Position = 1,
+            HelpMessage = "Number of the month (MM) to report usage")]
+        [ValidatePattern("^(0[1-9]|1[0-2])$")]
+        public string UsageMonth { get; set; }
+
+        [Parameter(Mandatory = false,
+            Position = 2,
+            HelpMessage = "Year (YYYY) to report usage")]
+        [ValidatePattern(@"^(19|20)\d\d$")]
+        public string UsageYear { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            RemoteAppOperationStatus operationStatus = RemoteAppOperationStatus.Failed;
+            DateTime today = DateTime.Now;
+            CollectionUsageDetailsResult detailsUsage = null;
+            string locale = String.Empty;
+
+            if (String.IsNullOrWhiteSpace(UsageMonth))
+            {
+                UsageMonth = today.Month.ToString();
+            }
+
+            if (String.IsNullOrWhiteSpace(UsageYear))
+            {
+                UsageYear = today.Year.ToString();
+            }
+
+            locale = System.Globalization.CultureInfo.CurrentCulture.ToString();
+
+            detailsUsage = CallClient(() => Client.Collections.GetUsageDetails(CollectionName, UsageYear, UsageMonth, locale), Client.Collections);
+
+            while (true)
+            {
+                RemoteAppOperationStatusResult operationResult = CallClient(() => Client.OperationResults.Get(detailsUsage.UsageDetails.OperationTrackingId), Client.OperationResults);
+                if(operationResult.RemoteAppOperationResult.Status == RemoteAppOperationStatus.Success ||
+                    operationResult.RemoteAppOperationResult.Status == RemoteAppOperationStatus.Failed)
+                {
+                    operationStatus = operationResult.RemoteAppOperationResult.Status;
+                    break;
+                }
+
+                System.Threading.Thread.Sleep(1000);
```

Comment:
Why do you infinitely loop while the return status code is not Success or Failed? What if you will get constantly something different? (like Forbidden?) Then this code will never end. Please consider re-factoring it to exclude while(true) loop.